File: 프로그래머스/unrated/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.java
class Solution {
    public String solution(String myString, String pat) {
        String answer = "";
        while(myString.indexOf(pat) != -1){
            answer += myString.substring(0, myString.indexOf(pat) + pat.length());
            myString = myString.substring(myString.indexOf(pat) + pat.length());
            
        }
        return answer;
    }
}

File: 프로그래머스/unrated/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string myString, string pat) {
    string subMyString = "";
    while(myString.find(pat) != string::npos){
        subMyString += myString.substr(0,myString.find(pat) + pat.size());
        myString = myString.substr(myString.find(pat) + pat.size());
    }
    return subMyString;
}